middleware/auto panics on non existent directory

Error log:
```
2016/11/06 18:36:19 [WARNING] Directory does not exist: /etc/source/files/zones/
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x5dfbe1]

goroutine 1 [running]:
panic(0x1b02bc0, 0xc420014020)
	/home/miek/upstream/go/src/runtime/panic.go:500 +0x1a1
github.com/miekg/coredns/middleware/auto.(*Zones).Walk.func1(0xc42048aea0, 0x18, 0x0, 0x0, 0x27f44a0, 0xc4202d63f0, 0x1, 0xc420212cb0)
	/home/miek/g/src/github.com/miekg/coredns/middleware/auto/walk.go:26 +0x51
path/filepath.Walk(0xc42048aea0, 0x18, 0xc420212cb0, 0x0, 0xc4202d63c0)
	/home/miek/upstream/go/src/path/filepath/path.go:396 +0x8b
github.com/miekg/coredns/middleware/auto.(*Zones).Walk(0xc42037d2c0, 0xc42048aea0, 0x18, 0xc420424cec, 0x4, 0xc42044dc20, 0x0, 0x0, 0x0, 0x0, ...)
	/home/miek/g/src/github.com/miekg/coredns/middleware/auto/walk.go:65 +0x1da
github.com/miekg/coredns/middleware/auto.setup.func1(0x0, 0x0)
	/home/miek/g/src/github.com/miekg/coredns/middleware/auto/setup.go:34 +0x5f
github.com/mholt/caddy.startWithListenerFds(0x28046a0, 0xc420315480, 0xc4202fe270, 0x0, 0x64, 0x0)
	/home/miek/g/src/github.com/mholt/caddy/caddy.go:479 +0x458
github.com/mholt/caddy.Start(0x28046a0, 0xc420315480, 0x28046a0, 0xc420315480, 0x0)
	/home/miek/g/src/github.com/mholt/caddy/caddy.go:427 +0xff
github.com/miekg/coredns/coremain.Run()
	/home/miek/g/src/github.com/miekg/coredns/coremain/run.go:80 +0x16f
main.main()
	/home/miek/g/src/github.com/miekg/coredns/coredns.go:6 +0x14
```

My suggestion would be, that it emits a warning, but doesn't panic. Will pre-create the directory for now until it is populated by my sidecar container.

--- a/middleware/auto/walk.go
+++ b/middleware/auto/walk.go
@@ -23,7 +23,7 @@ func (a Auto) Walk() error {
 	}
 
 	filepath.Walk(a.loader.directory, func(path string, info os.FileInfo, err error) error {
-		if info.IsDir() {
+		if info == nil || info.IsDir() {
 			return nil
 		}
 

--- a/middleware/auto/walk_test.go
+++ b/middleware/auto/walk_test.go
@@ -52,6 +52,25 @@ func TestWalk(t *testing.T) {
 	}
 }
 
+func TestWalkNonExistent(t *testing.T) {
+	log.SetOutput(ioutil.Discard)
+
+	nonExistingDir := "highly_unlikely_to_exist_dir"
+
+	ldr := loader{
+		directory: nonExistingDir,
+		re:        regexp.MustCompile(`db\.(.*)`),
+		template:  `${1}`,
+	}
+
+	a := Auto{
+		loader: ldr,
+		Zones:  &Zones{},
+	}
+
+	a.Walk()
+}
+
 func createFiles() (string, error) {
 	dir, err := ioutil.TempDir(os.TempDir(), "coredns")
 	if err != nil {

